src/mds/Locker.cc

```diff
@@ -2234,6 +2285,11 @@ int Locker::issue_caps(CInode *in, Capability *only_cap)
 
     // add in any xlocker-only caps (for locks this client is the xlocker for)
     allowed |= xlocker_allowed & in->get_xlocker_mask(it->first);
+    if (in->is_dir()) {
+      allowed &= ~CEPH_CAP_ANY_DIR_OPS;
+      if (allowed & CEPH_CAP_FILE_EXCL)
+	allowed |= cap->get_lock_cache_allowed();
```

Comment:
nit: funky indentation here

Reply:
indentation is correct here. the 'allowed ...' line is a tab, the 'if (allowed ...' line is 6 spaces.

Reply:
Ok, thanks. github--